### PR TITLE
Support for different Thumbsizes

### PR DESCRIPTION
**Pull request by MikeW1986**
Added PlatformEffect for Android and redraw Seekbar for different Thumbsizes.
Example Slider is added.

### CODE REVIEW

**Comment by MikeW1986** [comment]
this pull request is depending on my question in https://github.com/halkar/xamarin-range-slider/issues/128

it is my first pull request ever, so if i am missing something or doing something wrong, please say it ;)

**Comment by halkar** [comment]
@MikeW1986 sorry for the delay. I need to review your PR and make sure it doesn't break compatibility.

File: Xamarin.RangeSlider.Droid/RangeSliderControl.cs [hunk 1]
```diff
@@ -908,14 +914,15 @@ protected override void OnRestoreInstanceState(IParcelable parcel)
...
-                buttonToDraw = ThumbDisabledImage;
-            else
-                buttonToDraw = pressed ? ThumbPressedImage : ThumbImage;
-            
-            canvas.DrawBitmap(buttonToDraw, screenCoord - _thumbHalfWidth, _textOffset, _paint);
+        {
+            Bitmap buttonToDraw;
+            if (!Enabled || (!ActivateOnDefaultValues && areSelectedValuesDefault)) {
+              buttonToDraw = ThumbDisabledImage;
+              canvas.DrawBitmap(buttonToDraw, screenCoord - ThumbDisabledImage.Width * 0.5f, _textOffset + (ThumbImage.Height - ThumbDisabledImage.Height) / 2, _paint);
```
Comment (halkar) [inline Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: Logic for all states should be the same.

File: Xamarin.RangeSlider.Droid/RangeSliderControl.cs [hunk 2]
```diff
@@ -759,8 +759,12 @@ protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
         /// </summary>
         protected override void OnDraw(Canvas canvas)
         {
-            base.OnDraw(canvas);
-
+            base.OnDraw(canvas);
+            if (ThumbImage != null) {
```
Comment (halkar) [inline Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: This code should only sit in `UpdateThumbImages`.

Comment (MikeW1986) [reply Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: the only idea i have is to make an new property "customThumbImage" and don't use ThumbImage in the PlattformEffect

        private Bitmap _customThumbImage;
        public Bitmap CustomThumbImage 
        {
            get { return _customThumbImage; }
            set {
                _customThumbImage = value;
                UpdateThumbImages();
            }
        }

and in UpdateThumbImages()

            ThumbImage = DefaultThumbImage ?? (CustomThumbImage ?? BitmapUtil.DrawableToBitmap(GetThumbDrawable()));

thx for feedback

File: Xamarin.RangeSlider.Droid/RangeSliderControl.cs [hunk 3]
```diff
@@ -353,7 +362,7 @@ private void Init(Context context, IAttributeSet attrs)
 
         private void UpdateThumbImages()
         {
-            ThumbImage = DefaultThumbImage ?? BitmapUtil.DrawableToBitmap(GetThumbDrawable());
+            ThumbImage = DefaultThumbImage ?? (CustomThumbImage ?? BitmapUtil.DrawableToBitmap(GetThumbDrawable()));
             ThumbPressedImage = DefaultThumbPressedImage ?? BitmapUtil.DrawableToBitmap(GetPressedThumbDrawable());
```
Comment (halkar) [inline Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: It should support all images.

File: Xamarin.RangeSlider.Forms.Samples/Xamarin.RangeSlider.Forms.Samples.Droid/PlattformEffects/RangeSliderEffect.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      var res = Resources.System;
+
+      Context context = Xamarin.Forms.Forms.Context;
+      Bitmap icon = BitmapFactory.DecodeResource(context.Resources, Resource.Drawable.scrubber_control_normal_holo);
+
+      var drawable = ContextCompat.GetDrawable(context, Resource.Drawable.scrubber_control_normal_holo);
+      var thumbImage = new BitmapDrawable(ctrl.ThumbImage);
+
+      ctrl.CustomThumbImage = ConvertToBitmap(drawable, icon.Width, icon.Height);
+      //ctrl.ThumbPressedImage = ConvertToBitmap(drawable, icon.Width, icon.Height);
```
Comment (halkar) [inline Xamarin.RangeSlider.Forms.Samples/Xamarin.RangeSlider.Forms.Samples.Droid/PlattformEffects/RangeSliderEffect.cs]: Remove commented code.

File: Xamarin.RangeSlider.Droid/RangeSliderControl.cs [hunk 5]
```diff
@@ -224,6 +224,15 @@ public bool MaxThumbHidden
 
         public Bitmap ThumbDisabledImage { get; set; }
         public Bitmap ThumbImage { get; set; }
+        private Bitmap _customThumbImage;
+        public Bitmap CustomThumbImage
```
Comment (halkar) [inline Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: I don't like the idea of having three different properties for each image. It's better to split `UpdateThumbImages` into two functions.

**Comment by MikeW1986** [comment]
what do you think about calculating _thumbHalfWidth and _thumbHalfHeight depending on the biggest Thumb ( Pressed, Disabled...) because it could be, that the pressed thumb is bigger than der normal Thumb?

**Comment by halkar** [comment]
@MikeW1986 all thumbs should be of the same size. I don't want to overcomplicate.

File: Xamarin.RangeSlider.Droid/RangeSliderControl.cs [hunk 6]
```diff
@@ -562,6 +567,22 @@ public void SetThumbShadowPath(Path thumbShadowPath)
...
 
+        public void SetCustomThumbImage(Bitmap customThumbImage) 
+        {
+            ThumbImage = customThumbImage;
+            UpdateThumbValues();
+        }
+
+        public void SetCustomThumbPressedImage(Bitmap customThumbImage) 
+        {
+            ThumbPressedImage = customThumbImage;
```
Comment (halkar) [inline Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: `UpdateThumbValues` should be here and in the next method as well as there is no specific order in which these methods must be called.

Comment (MikeW1986) [reply Xamarin.RangeSlider.Droid/RangeSliderControl.cs]: ok done

**Comment by halkar** [comment]
I'll merge your changes and release the new version later this week.

**Pull request merged**